src/Illuminate/Database/Schema/Blueprint.php

```diff
@@ -1102,11 +1102,15 @@ public function boolean($column)
      * Create a new enum column on the table.
      *
      * @param  string  $column
-     * @param  array  $allowed
+     * @param  string|array  $allowed
      * @return \Illuminate\Database\Schema\ColumnDefinition
      */
-    public function enum($column, array $allowed)
+    public function enum($column, string|array $allowed)
     {
+        if (is_string($allowed) && enum_exists($allowed)) {
+            $allowed = array_column($allowed::cases(), 'value');
```

Comment:
What if enum is not a BackedEnum but a UnitEnum?

Reply:
Yes, you are right, I forgot to add the line for handling UnitEnum, but as per the existing cast attribute to handle the UnitEnum https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Casts/AsEnumArrayObject.php#L46 

I think this way is the best approach for this proposal, adding a ternary value like this:

```php
/**
     * Create a new enum column on the table.
     *
     * @param  string  $column
     * @param  string|array  $allowed
     * @return \Illuminate\Database\Schema\ColumnDefinition
     */
    public function enum($column, string|array $allowed)
    {
        if (is_string($allowed) && enum_exists($allowed)) {
            $allowed = array_column(
                $allowed::cases(),
                is_subclass_of($allowed, BackedEnum::class)
                    ? 'value'
                    : 'name'
            );
        }

        return $this->addColumn('enum', $column, compact('allowed'));
    }
```

What do you think?